src/inference.py

```diff
@@ -68,10 +73,14 @@
         detector_tiled_api_key=io_parameters.detector_tiled_api_key,
     )
 
-    # Load model
-    model = Autoencoder.load_from_checkpoint(
-        f"{io_parameters.models_dir}/{io_parameters.uid_retrieve}/last.ckpt"
-    )
+    # Load latest model from the registry
+    if hasattr(io_parameters, "mlflow_model") and io_parameters.mlflow_model:
+        model_name = io_parameters.mlflow_model
+    else:
+        model_name = io_parameters.uid_retrieve
+    logger.info(f"Loading latest model from MLflow registry: {model_name}")
+
+    model = mlflow.pytorch.load_model(f"models:/{model_name}/latest")
```

Comment:
Does `model_name` need to be a unique identifier? Do you know what happens if 2 different models are registered with the same name?